help/search/_posts/2021-04-26-author-search.md

```diff
@@ -0,0 +1,107 @@
+
+Next, we'll use all of the name variants discovered so far, and take one step further -- we'll generate possible name variations, so for example authors with more than 2 names `zeta john jones` -- they could be known in the scientific literature as `zeta, j`, `zeta john, j`, and even `zeta, j j` (well, and possibly `john jones, zeta` -- if we can't tell what is the first and last name, you get the idea...)
+
+So the parser will tentatively generate a multitude of possible names and spellings and will use them all to search for other 'real' people. If it finds anybody new, it will sigh deeply, discard the temporary variants and then proceed to generate new `transliterations` and new `synonyms` of any new (real) person that was found in ADS database. As a way of example, this is what we'll get:
+
+- adamzuk, peter
+- adamčuk, peter
+- adamčuková, petra
+- stoklasová, petra
+- stoklasova, petra
+
+All these names were added because our parser consulted knowledge base of author names. Our parser doesn't know this is one and the same person; but it has a way of seeing/discovering the chain of name changes. If the knowledge database is correct, which is most of the time, it will be able to start from the user input and arrive at other names. 
+
+In the next stages, we'll deduplicate entries and normalize them once again. We'll also generate few other different queries to account for idiosynracies in how ADS stores/searches/curates data. For example, oftentimes the papers are given to ADS without full author names and will only contain initials, so even though you may want to search for the full name, we'll also (automatically) expand the search to include the initials (or vice versa - you are searching for initials but we'll insist to expand your search -- be less precise, more comprehensive). It depends on the input. If for example user typed `acamczuk, p` then our parser will assume they are fine with `adamczuk, p*` (and give them all peters, piotrs, pauls, pans and pons), but if the user typed `adamczuk, peter` then the search engine will take care not to search for `adamczuk, p*`, instead it will return `adamczuk, p`, `adamczuk, p <some other name>` and all of the synonyms thereof (`adamčuková, p`, `adamčuková, p *`, `stoklasová, p`...)
+
+So what started as few characters (one query) is turned behind the scenes into a massive and very complicated query. There is a constant battle between the desire to be precise, and the desire to include everybody who should be there (recall). Yet all of this without really knowing much about the person, all we have is just a bunch of characters. The logic is complex, it takes advantage of what ADS knows about the authors and includes a knowledge base (plus business logic) but it is far from being able to understand 'intentions' of the user; and it has no real model of a person (and their relationships).
```

Comment:
* "So what started as few characters" -> "So what started as **a** few characters"
* "it is far from being able to understand 'intentions' of the user" -> "it is far from being able to understand **the** 'intentions' of the user"